### PR TITLE
Show a more real example in data collectors doc

### PR DESCRIPTION
**Pull request by wouterj**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | 2.3+ |
| Fixed tickets | - |

### CODE REVIEW

File: cookbook/profiler/data_collector.rst [hunk 1]
```diff
@@ -16,70 +16,71 @@ Creating a custom data collector is as simple as implementing the
 
     interface DataCollectorInterface
     {
-        /**
-         * Collects data for the given Request and Response.
-         *
-         * @param Request    $request   A Request instance
-         * @param Response   $response  A Response instance
-         * @param \Exception $exception An Exception instance
-         */
```
Comment (WouterJ) [inline cookbook/profiler/data_collector.rst]: Removed the PHPdoc as I don't think it's very usefull (short descriptions are better described directly after the code example and the typehint already shows which instances are passed).

File: cookbook/profiler/data_collector.rst [hunk 2]
```diff
@@ -16,70 +16,71 @@ Creating a custom data collector is as simple as implementing the
...
 
-    class MyCollector extends DataCollector
+    class RequestCollector extends DataCollector
     {
         public function collect(Request $request, Response $response, \Exception $exception = null)
         {
             $this->data = array(
-                'variable' => 'value',
+                'method' => $request->getMethod(),
+                'acceptable_content_types' => $request->getAcceptableContentTypes(),
```
Comment (WouterJ) [inline cookbook/profiler/data_collector.rst]: decided to go for a more real usage, to show a way to collect data and how to access collected data in the template. On top of that, using real use-case examples, things are often easier to understand.

File: cookbook/profiler/data_collector.rst [hunk 3]
```diff
@@ -16,70 +16,71 @@ Creating a custom data collector is as simple as implementing the
...
         public function getName()
         {
-            return 'app.my_collector';
+            return 'app.request_collector';
         }
     }
 
+The getters are added to give the template access to the collected information.
+
+.. caution::
```
Comment (WouterJ) [inline cookbook/profiler/data_collector.rst]: Moved the caution some lines below, as it was talking about overriding a `serialize()` method which part of the abstract data collector (introduced just above).

File: cookbook/profiler/data_collector.rst [hunk 4]
```diff
@@ -16,70 +16,71 @@ Creating a custom data collector is as simple as implementing the
...
 
-The value returned by ``getName()`` must be unique in the application. This value
-is also used to access the information later on (see :doc:`/cookbook/testing/profiling`
-for instance).
-
-The ``collect()`` method is responsible for storing the collected data in local
-properties.
+The
+:method:`Symfony\\Component\\HttpKernel\\DataCollector\\DataCollectorInterface::getName`
+method returns the name of the data collector and must be unique in the
```
Comment (OskarStark) [inline cookbook/profiler/data_collector.rst]: ... ``unique within`` ... ?

**Comment by OskarStark** [comment]
Status: Reviewed

File: cookbook/profiler/data_collector.rst (hunk 4 above)
Comment (stof) [inline cookbook/profiler/data_collector.rst]: unique within data collector names. But yeah, the message could be improved

File: cookbook/profiler/data_collector.rst (hunk 2 above)
Comment (javiereguiluz) [inline cookbook/profiler/data_collector.rst]: I agree completely!

File: cookbook/profiler/data_collector.rst [hunk 5]
```diff
@@ -139,24 +142,25 @@ block and set the value of two variables called ``icon`` and ``text``:
...
-                <span>100 units</span>
+                <b>Method</b>
+                <span>{{ collector.method }}</span>
             </div>
+
             <div class="sf-toolbar-info-piece">
-                <b>Another piece of data</b>
-                <span>300 units</span>
+                <b>Accepted content type</b>
+                <span>{{ collector.acceptablecontenttypes|join(', ') }}</span>
```
Comment (xabbuh) [inline cookbook/profiler/data_collector.rst]: `collector.acceptableContentTypes` is more readable

**Comment by weaverryan** [comment]
This is *so* much better. Thanks for fixing my ridiculous example from many years ago - real examples ftw!

File: cookbook/profiler/data_collector.rst (hunk 5 above)
Comment (weaverryan) [inline cookbook/profiler/data_collector.rst]: fixed!

**Pull request merged**